### PR TITLE
feat(clerk-js,types): Add protect to billing components

### PR DESCRIPTION
**Pull request by nikospapcom**
## Description

In this pr we are adding protect to billing components
⚠️ merge this pr after fapi/dapi introduce `org:sys_billing:read` & `org:sys_billing:read`

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ type BillingProviderValue = {
   currentPage: BillingPages;
   goToPlanAndBilling: () => void;
   goToManageBillingPlan: () => void;
+  checkPermissions?: boolean;
```
Comment (AlexNti) [inline packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx:20]: 💯

**Review by AlexNti — APPROVED** [review]

File: packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx (hunk 1 above)
Comment (panteliselef) [inline packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx:20]: I imagine that this is needed to us to support Billing inside the UserProfile where permissions do not exist, right ?

File: packages/clerk-js/src/ui/common/Billing/PlanAndBillingScreen.tsx [hunk 2]
```diff
@@ -139,21 +140,36 @@ const CurrentPlanSection = () => {
...
-          onClick={goToManageBillingPlan}
-          id='currentPlan'
-          localizationKey={
-            currentPlan.priceInCents
-              ? localizationKeys('billing.currentPlanSection.primaryButton')
-              : localizationKeys('billing.currentPlanSection.primaryButton__upgrade')
-          }
-        />
+        {checkPermissions ? (
+          <Protect permission='org:sys_billing:manage'>
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/common/Billing/PlanAndBillingScreen.tsx:144]: Maybe using `useProtect` makes more sense now that we have `checkPermissions`

**Review by panteliselef — APPROVED** [review]

File: packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx (hunk 1 above)
Comment (nikospapcom) [reply packages/clerk-js/src/ui/common/Billing/BillingProvider.tsx:20]: Correct @panteliselef

**Pull request merged**